examples/collada.js

```diff
@@ -0,0 +1,89 @@
+/* global itowns, document, renderer, GuiTools, Promise */
+// # Simple Globe viewer
+
+// Define initial camera position
+// Coordinate can be found on https://www.geoportail.gouv.fr/carte
+// setting is "coordonnée geographiques en degres decimaux"
+
+// Position near Gerbier mountain.
+var positionOnGlobe = { longitude: 4.21655, latitude: 44.84415, altitude: 1500 };
+
+// `viewerDiv` will contain iTowns' rendering area (`<canvas>`)
+var viewerDiv = document.getElementById('viewerDiv');
+
+// Instanciate iTowns GlobeView*
+var globeView = new itowns.GlobeView(viewerDiv, positionOnGlobe, { renderer: renderer });
+
+var menuGlobe = new GuiTools('menuDiv');
+
+var promiseElevation = [];
+
+menuGlobe.view = globeView;
+
+function addLayerCb(layer) {
+    return globeView.addLayer(layer).then(function addGui() {
+        var la = globeView.getLayers(function cl(l) { return l.id === layer.id; })[0];
```

Comment:
`whenReady` is resolved with the layer, so you already have it as arguments to your `addGui` function above.

Reply:
I tried with whenReady, but I was surprised that it did not work. It'll look for this bug

Reply:
```js
function addLayerCb(layer) {
    return globeView.addLayer(layer).then(function addGui(la) {
        if (la.type === 'color') {
            menuGlobe.addImageryLayerGUI(la);
        } else if (la.type === 'elevation') {
            menuGlobe.addElevationLayerGUI(la);
        }
    });
}
```
that's what I meant.

Reply:
whenReady is returned by `addLayer`

Reply:
yes, i know, as I told you I tried but it's buggy. I will create a issue

Reply:
This snippet works, I've tested.

Reply:
ok, I didn't follow, I updated